src/utils/codec.ts

```diff
@@ -0,0 +1,318 @@
+            Array.from(str, (byte: string) => {
+                const index = scheme.table.indexOf(byte)
+
+                // return empty string if it is tail
+                if (index === -1) {
+                    return ''
+                }
+
+                return ('0'.repeat(scheme.bits - 1) + Number(index).toString(2)).slice(-scheme.bits)
+            }).reduce((o: string, c: string) => (o += c)),
+        )
+    }
+
+    /**
+     * Build `Codec` from base64 string
+     *
+     * @param {string} b64 - base64 string
+     */
+    static fromBase64(b64: string): Codec {
+        return Codec.from(BASE64, b64)
+    }
+
+    /**
+     * Build `Codec` from hex string
+     *
+     * @param {string} str - hex string
+     */
+    static fromHex(hex: string): Codec {
+        return Codec.from(HEX, hex)
+    }
+
+    /**
+     * Build Codec from utf-16 string
+     *
+     * @param {string} str - utf16 string
+     */
+    static fromUtf16(str: string): Codec {
+        return new Codec(
+            Array.from(str, (char: string) => {
+                const point = char.codePointAt(0)
+                if (point === undefined) {
+                    return '0'.repeat(8)
+                }
+
+                // Complete the 0 prefix from every byte
+                return ('0'.repeat(8) + point.toString(2)).slice(-8)
+            }).reduce((o: string, c: string) => (o += c)),
+        )
+    }
+
+    /**
+     * Convert code point into target radix string
+     *
+     * @name P2RS - Point to Radix String
+     *
+     * @param {CodecScheme} scheme      - target codec scheme
+     * @param {Point}       point       - code point
+     * @param {string}      str         - the output string
+     * @param {boolean}     completion  - if complete the prefix
+     */
+    static P2RS(scheme: CodecScheme, point: Point, str: string, completion: boolean): [number, string] {
+        if (point === undefined) {
+            return [0, '']
+        } else if (point === 0) {
+            return [0, str]
+        }
+
+        // Complete prefix if it is required
+        const reminder = completion
+            ? (scheme.table[0].repeat(scheme.bits - 1) + scheme.table[point & ((1 << scheme.bits) - 1)]).slice(
+                  -scheme.bits,
+              )
+            : scheme.table[point & ((1 << scheme.bits) - 1)]
+
+        // Recursive result
+        return this.P2RS(scheme, point >> scheme.bits, reminder + str, completion)
+    }
+
+    /**
+     * Convert code points into target radix string
+     *
+     * @name Ps2RS - Points to Radix String
+     *
+     * @param {CodecScheme} scheme     - target codec scheme.
+     * @param {Point[]}     points     - code points
+     * @param {boolean}     completion - complete prefix
+     */
+    static Ps2RS(scheme: CodecScheme, points: Point[], completion: boolean): string {
+        return points.map((point: Point) => Codec.P2RS(scheme, point, '', completion)[1]).reduce((o, c) => (o += c))
+    }
```

Comment:
I'll delete these code, because it seems they are useless now —— Wrong caculating way in the commit before last.

Reply:
`Ps2RS` can caculating all coding formats from input string to target scheme, it is based on `Points`.

Using `binary` as base elements(current solution) is easy and better than that.